### ISSUE TITLE
Setting the job shell doesn't work as expected

### ISSUE THREAD

**Issue by rmb938** [Azure/brigade#756]
Output of `brig version`: ```v0.19.0```

Output of `kubectl version`: 
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.1", GitCommit:"eec55b9ba98609a46fee712359c7b5b365bdd920", GitTreeState:"clean", BuildDate:"2018-12-13T10:39:04Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.1", GitCommit:"eec55b9ba98609a46fee712359c7b5b365bdd920", GitTreeState:"clean", BuildDate:"2018-12-13T10:31:33Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```

When setting the job shell via the shell variable

```
j = new Job("sops", "mozilla/sops:latest")
j.shell = "/bin/bash"
```

this only changes the `#!` in the hook script.

However being the the container's command is `/bin/sh /hook/main.sh` the `#!` is ignored and is ran via `/bin/sh` instead. 

Typically the `#!` in a script is only used when executing directly i.e `./hook/main.sh`.

**Comment by radu-matei** [comment]
> However being the the container's command is `/bin/sh /hook/main.sh` the `#!` is ignored and is ran via `/bin/sh` instead.

Yes, it turns out in the end the shebang is ignored, because we use `bin/sh` directly:

[`c1.command = ["/bin/sh", "/hook/main.sh"];`](https://github.com/Azure/brigade/blob/master/brigade-worker/src/k8s.ts#L887)

So we can either:

- pass `job.shell` to the [`newRunnerPod`](https://github.com/Azure/brigade/blob/master/brigade-worker/src/k8s.ts#L866) (the function where we set the container command) , or:

- directly `exec` the script using the shebang defined in the script.

I'm trying to think if there are any differences between the two approaches, but couldn't find any yet.
Thoughts?

**Comment by rmb938** [comment]
Well I don't think we can execute it directly since there is no way to make that file executable since it comes from a secret (I could be wrong though).

So passing `job.shell` to the newRunnerPod probably would be ideal.

**Comment by radu-matei** [comment]
@rmb938 - fix in #759.
Let me know if it solves your use case.

Thanks!
